src/lib/types/bible.ts

```diff
@@ -49,6 +49,10 @@ export interface FrontendBibleBook extends BaseBible {
     content: { chapters: FrontendChapterContent[] } | null;
 }
 
+export interface FrontendBibleBookWithLanguageCode extends FrontendBibleBook {
+    languageCode: string;
+}
```

Comment:
Is this needed? Looking below it seems like the type is applied but might not actually be needed.

Reply:
In other words I'm wondering if TS can infer it based on the function's return.

Reply:
When language code was used, the linter complained that it was absent.

Reply:
Ah gotcha, looks like it's because the sort function takes two types so the linter doesn't know what will be returned. Playing around with it this function signature fixed it, but it's probably worse than what you've got 😆 
```
export function sortAndFilterBibles<T extends { abbreviation: string; name: string; languageId: number }>(
    bibles: T[],
    currentLanguageInfoId: number | undefined,
    searchQuery: string
) {
```